### PR TITLE
ceph: only try to build specific provisioner

### PR DESCRIPTION
**Pull request by leseb**
Build only on the provisioner that was discovered.

Signed-off-by: Sébastien Han <seb@redhat.com>

[test ceph]

### CODE REVIEW

File: tests/scripts/makeTestImages.sh [hunk 1]
```diff
@@ -1,37 +1,48 @@
-#!/bin/bash -e
+#!/bin/bash
+set -ex
 
 scriptdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 # shellcheck disable=SC1090
 source "${scriptdir}/../../build/common.sh"
 
+# list of images to build
+imagesToBuild=(ceph cockroachdb cassandra nfs yugabytedb)
```
Comment (travisn) [inline tests/scripts/makeTestImages.sh:12]:
```suggestion
imagesToBuild=(ceph cockroachdb cassandra edgefs nfs yugabytedb)
```

File: Jenkinsfile [hunk 2]
```diff
@@ -101,7 +101,7 @@ pipeline {
                 sh 'build/run make -j\$(nproc) mod.check'
                 sh 'git diff-index --quiet HEAD || { echo "CHANGES FOUND. You may need to run make clean"; git status -s; exit 1; }'
                 // run the build
-                sh 'build/run make -j\$(nproc) build.all'
+                sh (script: "build/run make -j\$(nproc) IMAGES=${env.testProvider} build", returnStdout: true)
```
Comment (travisn) [inline Jenkinsfile:104]: If it's not a PR we still need to run `build.all`. A big part of the savings we are seeing with this change is skipping the arm64 build, but master and release builds can't skip that.

Comment (travisn) [reply Jenkinsfile:104]: `env.isOfficialBuild` should be useful to differentiate PR builds from master/release.

File: Jenkinsfile [hunk 3]
```diff
@@ -101,7 +101,13 @@ pipeline {
                 sh 'build/run make -j\$(nproc) mod.check'
                 sh 'git diff-index --quiet HEAD || { echo "CHANGES FOUND. You may need to run make clean"; git status -s; exit 1; }'
                 // run the build
-                sh 'build/run make -j\$(nproc) build.all'
+                script {
+                    if (env.isOfficialBuild == "false") {
+                        sh (script: "build/run make -j\$(nproc) IMAGES=${env.testProvider} build", returnStdout: true)
+                    } else {
+                        sh (script: "build/run make -j\$(nproc) build", returnStdout: true)
```
Comment (travisn) [inline Jenkinsfile:108]: Official builds still need arm64.
```suggestion
                        sh (script: "build/run make -j\$(nproc) build.all", returnStdout: true)
```

File: Jenkinsfile [hunk 4]
```diff
@@ -101,7 +101,13 @@ pipeline {
                 sh 'build/run make -j\$(nproc) mod.check'
                 sh 'git diff-index --quiet HEAD || { echo "CHANGES FOUND. You may need to run make clean"; git status -s; exit 1; }'
                 // run the build
-                sh 'build/run make -j\$(nproc) build.all'
+                script {
+                    if (env.isOfficialBuild == "false") {
+                        sh (script: "build/run make -j\$(nproc) IMAGES=${env.testProvider} build", returnStdout: true)
```
Comment (travisn) [inline Jenkinsfile:106]: What if there is no testProvider detected? It will just build all, right?

File: tests/scripts/makeTestImages.sh [hunk 5]
```diff
@@ -1,37 +1,50 @@
...
 
+# list of images to build
+imagesToBuild=(ceph cockroachdb cassandra edgefs nfs yugabytedb)
 
 case "${1:-}" in
     save)
+    # Storage backend to tag/save/build
+    storageBackend="$3"
+    if [ -z "$storageBackend" ]; then
+        storageBackend="all"
```
Comment (travisn) [inline tests/scripts/makeTestImages.sh:19]: If it's not set, why not leave it as the empty string?

File: tests/scripts/makeTestImages.sh [hunk 6]
```diff
@@ -1,37 +1,50 @@
...
+
+# readable bash output
+export PS4='+${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
 
 scriptdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 # shellcheck disable=SC1090
 source "${scriptdir}/../../build/common.sh"
 
+# list of images to build
+imagesToBuild=(ceph cockroachdb cassandra edgefs nfs yugabytedb)
```
Comment (travisn) [inline tests/scripts/makeTestImages.sh:12]: I know I suggested to add edgefs to the list, but now I see that it wasn't previously found anywhere in this file. For consistency, let's not add it to the list, and separately investigate the implications of it not being in this list.

File: Jenkinsfile (hunk 4 above)
Comment (leseb) [reply Jenkinsfile:106]: I added a check for that to build all.

File: Jenkinsfile (hunk 3 above)
Comment (leseb) [reply Jenkinsfile:108]: It wasn't there previously but ok...

File: tests/scripts/makeTestImages.sh (hunk 5 above)
Comment (leseb) [reply tests/scripts/makeTestImages.sh:19]: I just think it's easier to give it a proper value, it makes the code more understandable later.

**Comment by leseb** [comment]
Abandoning this for now as I have no time to work on this. Also extracting a small piece in https://github.com/rook/rook/pull/6218 was a good start already.

**Pull request closed without merging**